src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -45,6 +45,74 @@ impl ScalarFunc {
         }
     }
 
+    #[inline]
+    pub fn oct_int<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        match self.children[0].eval_int(ctx, row) {
+            Err(_e) => self.oct_string(ctx, row),
+            Ok(None) => Ok(None),
+            Ok(Some(res)) => Ok(Some(Cow::Owned(format!("{:o}", res).into_bytes()))),
+        }
+    }
+
+    #[inline]
+    pub fn oct_string<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let input = try_opt!(self.children[0].eval_string(ctx, row));
+        let mut num: u64 = 0;
+        let mut i: usize = 1;
+        let mut overflow = false;
+        let neg = match input[0] {
+            b'-' => true,
+            b'+' => false,
+            _ => {
+                i -= 1;
+                false
+            }
+        };
+
+        while i < input.len() {
```

Comment:
Maybe you can use https://doc.rust-lang.org/std/primitive.str.html#method.parse to parse the string to u64

and deal with negative or overflow like this: https://github.com/pingcap/tidb/blob/master/expression/builtin_string.go#L2533

Reply:
Hi! but how convert the Vec  to String and then can I use the parse.